classes/class-wc-rest-connect-stripe-oauth-connect-controller.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( class_exists( 'WC_REST_Connect_Stripe_Oauth_Connect_Controller' ) ) {
+	return;
+}
+
+class WC_REST_Connect_Stripe_Oauth_Connect_Controller extends WC_REST_Connect_Base_Controller {
+	protected $rest_base = 'connect/stripe/oauth/connect';
+	private $stripe;
+
+	public function __construct( WC_Connect_Stripe $stripe, WC_Connect_API_Client $api_client, WC_Connect_Service_Settings_Store $settings_store, WC_Connect_Logger $logger ) {
+		parent::__construct( $api_client, $settings_store, $logger );
+		$this->stripe = $stripe;
+	}
+
+	public function post( $request ) {
+		$data = $request->get_json_params();
+
+		$response = $this->stripe->connect_oauth( $data['state'], $data['code'] );
+
+		if ( is_wp_error( $response ) ) {
+			$this->logger->debug( $response, __CLASS__ );
+			return new WP_REST_Response( array(
+				'success'   => false,
+				'data'      => array(
+					'message' => $response->get_error_message(),
+				),
+			), 400 );
+		}
+
+		return new WP_REST_Response( array(
```

Comment:
It's enough to `return` the `array`, it's wrapped into a `WP_REST_Response(200)` by default.